admin/database/postgres/postgres.go

```diff
@@ -840,36 +781,126 @@ func (c *connection) FindProjectInvite(ctx context.Context, projectID, userEmail
 	res := &database.ProjectInvite{}
 	err := c.getDB(ctx).QueryRowxContext(ctx, "SELECT * FROM project_invites WHERE lower(email) = lower($1) AND project_id = $2", userEmail, projectID).StructScan(res)
 	if err != nil {
-		return nil, parseErr(err)
+		return nil, parseErr("project invite", err)
 	}
 	return res, nil
 }
 
 func (c *connection) InsertProjectInvite(ctx context.Context, email, projectID, roleID, invitedByID string) error {
 	_, err := c.getDB(ctx).ExecContext(ctx, "INSERT INTO project_invites (email, invited_by_user_id, project_id, project_role_id) VALUES ($1, $2, $3, $4)", email, invitedByID, projectID, roleID)
 	if err != nil {
-		return parseErr(err)
+		return parseErr("project invite", err)
 	}
 	return nil
 }
 
 func (c *connection) DeleteProjectInvite(ctx context.Context, id string) error {
-	_, err := c.getDB(ctx).ExecContext(ctx, "DELETE FROM project_invites WHERE id = $1", id)
+	res, err := c.getDB(ctx).ExecContext(ctx, "DELETE FROM project_invites WHERE id = $1", id)
+	return checkDeleteRow("project invite", res, err)
+}
+
+func checkUpdateRow(target string, res sql.Result, err error) error {
+	if err != nil {
+		return parseErr(target, err)
+	}
+	n, err := res.RowsAffected()
+	if err != nil {
+		return parseErr(target, err)
+	}
+	if n == 0 {
+		return parseErr(target, sql.ErrNoRows)
+	}
+	if n > 1 {
+		// This should never happen
+		panic(fmt.Errorf("expected to update 1 row, but updated %d", n))
+	}
+	return nil
+}
+
+func checkDeleteRow(target string, res sql.Result, err error) error {
 	if err != nil {
-		return parseErr(err)
+		return parseErr(target, err)
+	}
+	n, err := res.RowsAffected()
+	if err != nil {
+		return parseErr(target, err)
+	}
+	if n == 0 {
+		return parseErr(target, sql.ErrNoRows)
+	}
+	if n > 1 {
+		// This should never happen
+		panic(fmt.Errorf("expected to delete 1 row, but deleted %d", n))
 	}
 	return nil
 }
 
-func parseErr(err error) error {
+func parseErr(target string, err error) error {
 	if err == nil {
 		return nil
 	}
 	if errors.Is(err, sql.ErrNoRows) {
-		return database.ErrNotFound
+		if target == "" {
+			return database.ErrNotFound
+		}
+		return &wrappedError{
+			msg: fmt.Sprintf("%s not found", target),
```

Comment:
IMO the `not found` message should be generated separately in each DB method. Agreed it will mean extra code but messages would be better. 
Currently it adds same message for different where clauses.

Reply:
See earlier comment – there's no way to do this at the granularity of params without hurting queries. These kinds of extra checks should be at the API level.